File: JulyCircuit/special_shop.c
#include<stdio.h>
int main(){
	int testCase,sum,count,smallest;
	scanf("%d",&testCase);
	while(testCase!=0){
		int n,a,b;
		scanf("%d",&n);
		scanf("%d",&a);
		scanf("%d",&b);
		count=(a+b)-1;
		smallest=(a*(count*count)+b*((n-count)*(n-count)));
		while(count!=((n+a+b)-count)){
			sum=(a*(count*count)+b*((n-count)*(n-count)));
			if(sum<smallest)
			 	smallest=sum;
			count+=1;
		}
		printf("%d\n",smallest);
		testCase--;
	}
}

File: JulyCircuit/special_shop.py
import math

testCase = int(input())
while testCase != 0:
    pass
    n, a, b = input().split()
    sum_list = []

    n = int(n)
    a = int(a)
    b = int(b)
  
    approx_range = ((b*n)/(a+b))
    
    s1 = math.ceil(approx_range)
    s2 = math.floor(approx_range)
    s1 = a*pow(s1, 2)+b*pow((n-s1), 2)
    s2 = a*pow(s2, 2)+b*pow((n-s2), 2)
    if s1 < s2:
        smallest = s1
    else:
        smallest =s2
  

    print(smallest)
 
    testCase -= 1
